addons/account_asset/account_asset.py

```diff
@@ -400,51 +407,92 @@ def _get_move_check(self):
+                company_currency = line.asset_id.company_id.currency_id
+                current_currency = line.asset_id.currency_id
+                amount = current_currency.compute(line.amount, company_currency)
+                sign = (line.asset_id.category_id.journal_id.type == 'purchase' or line.asset_id.category_id.journal_id.type == 'sale' and 1) or -1
+                asset_name = line.asset_id.name + ' (%s/%s)' % (line.sequence, line.asset_id.method_number)
+                reference = line.asset_id.code
+                journal_id = line.asset_id.category_id.journal_id.id
+                partner_id = line.asset_id.partner_id.id
+                categ_type = line.asset_id.category_id.type
+                debit_account = line.asset_id.category_id.account_asset_id.id
+                credit_account = line.asset_id.category_id.account_depreciation_id.id
+                move_line_1 = {
+                    'name': asset_name,
+                    'account_id': credit_account,
+                    'debit': 0.0,
+                    'credit': amount,
+                    'journal_id': journal_id,
+                    'partner_id': partner_id,
+                    'currency_id': company_currency != current_currency and current_currency.id or False,
+                    'amount_currency': company_currency != current_currency and - sign * line.amount or 0.0,
+                    'analytic_account_id': line.asset_id.category_id.account_analytic_id.id if categ_type == 'sale' else False,
+                    'date': depreciation_date,
+                }
+                move_line_2 = {
+                    'name': asset_name,
+                    'account_id': debit_account,
+                    'credit': 0.0,
+                    'debit': amount,
+                    'journal_id': journal_id,
+                    'partner_id': partner_id,
+                    'currency_id': company_currency != current_currency and current_currency.id or False,
+                    'amount_currency': company_currency != current_currency and sign * line.amount or 0.0,
+                    'analytic_account_id': line.asset_id.category_id.account_analytic_id.id if categ_type == 'purchase' else False,
+                    'date': depreciation_date,
+                }
+                move_vals = {
+                    'ref': reference,
+                    'date': depreciation_date or False,
+                    'journal_id': line.asset_id.category_id.journal_id.id,
+                    'line_ids': [(0, 0, move_line_1), (0, 0, move_line_2)],
+                    'asset_id': line.asset_id.id,
+                    }
+                move = self.env['account.move'].create(move_vals)
+                line.write({'move_id': move.id, 'move_check': True})
+                created_moves |= move
+        elif group_entries and self:
+            # We can suppose that all lines have the same category
```

Comment:
If grouped, this method is either called : 
 - with all assets without category (l.122)
 - with all assets with the same category (l.126)